boto/exception.py

```diff
@@ -583,3 +583,18 @@ class InvalidInstanceMetadataError(Exception):
     def __init__(self, msg):
         final_msg = msg + '\n' + self.MSG
         super(InvalidInstanceMetadataError, self).__init__(final_msg)
+
+
+class BaseEndpointResolverError(Exception):
```

Comment:
Since this is "shared" code I would not be in favor of removing them in case we have to backport fixes. I could see moving them in with the `EndpointResolver` itself, is that what you had in mind?